prereise/gather/request_util.py

```diff
@@ -69,16 +69,16 @@ def retry(
 
     def decorator(func):
         limiter = RateLimit(interval)
-        func.retry_count = 0
 
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
+            wrapper.retry_count = 0
             for i in range(max_attempts):
```

Comment:
Just curious about the logic of the for loop here, what is the idea behind it?

Reply:
Which part do you mean? The function attribute, overall retry logic, etc? Just want to make sure I answer the right question

Reply:
Since we are not using the loop variable `i` here and it seems `wrapper.retry_count` also serves as a counter, I'm wondering why we need a `for` loop here rather than simply do `while wrapper.retry_count<=max_attempts`?

Reply:
Ah I see. The `while` loop would also work, but the distinction is the `wrapper.retry_count` variable is exposed to the decorated function (e.g. `download` in the `noaa_api` module) so while unusual, it could be modified which could break the loop. Compared to the `for` loop which is a bit more foolproof.

Reply:
The idea is that the user of the retry decorator shouldn't control the flow, that would defeat the point. The effect of exposing `wrapper.retry_count` does feel like leaking some details, but at least it's one-way (e.g. `download` doesn't modify the control flow defined by `retry`).

Reply:
I see. Good to know. Thanks.